splash/network_manager.py

```diff
@@ -26,10 +26,77 @@
 )
 from splash.response_middleware import ContentTypeMiddleware
 from splash import defaults
+from splash.qtutils import qt_header_items
 from splash.utils import to_bytes
 from splash.cookies import SplashCookieJar
 
 
+class _InvalidContentLength(ValueError):
+
+    def __init__(self, value):
+        if isinstance(value, bytes):
+            value = '0x' + value.hex()
+        message = 'Invalid Content-Length header value: {}'.format(value)
+        super().__init__(message)
+
+
+def _get_content_length(reply):
+    for name, value in qt_header_items(reply):
+        if bytes(name).lower() == b'content-length':
+            value = bytes(value).split(b',', 1)[0]
+            try:
+                value = value.decode('latin1')
+                value = int(value)
+            except (UnicodeDecodeError, ValueError):
+                raise _InvalidContentLength(value)
+            if value < 0:
+                raise _InvalidContentLength(value)
+            return value
+
+
+def _size_warrants_abort(sizes_and_sources, render_options, log, reply):
```

Comment:
this function looks over-complicated, can we improve it somehow?

Reply:
Maybe I can extract some of the inner logic into separate functions, such as the parsing of `max_size`.